src/goto-instrument/contracts/assigns.cpp

```diff
@@ -117,6 +117,8 @@ assigns_clauset::conditional_address_ranget::generate_snapshot_instructions()
     location));
 
   instructions.destructive_append(skip_program);
+
+  add_pragma_disable_assigns_check(instructions);
```

Comment:
Why do we need to disable inclusion check here?

Reply:
Because we don't need to check inclusion on definitions of CAR variables themselves,

Reply:
I think a comment here would be helpful to smooth debugging in the future.

Reply:
+100. I have no idea how this commit relates to the remainder of the PR. Does it perhaps touch code that was only introduced in this PR? If so, fold it into that commit. Where do I find tests? Why is this change being made at all?

Reply:
> I have no idea how this commit relates to the remainder of the PR.

The PR introduces write set checking in loop bodies. This commit ensures that each loop body is only instrumented once in presence of nested loops.

> Does it perhaps touch code that was only introduced in this PR?

Yes it improves the core checks introduced in `instrument loops per given assigns clauses` commit.

> If so, fold it into that commit.

I am not sure. This is an optimization commit. I intentionally kept it separate from the core commit that introduces checks. I thought it would make reviewing + debugging any potential issues in the future easier this way, but I could fold it into that commit if you think I should do that.

> Where do I find tests?

The test [loop-assigns-04] is included in `instrument loops per given assigns clauses` commit. This commit simply reduces the number of (repeated) checks generated in the output for that test.

I wasn't sure how to specify that in the test.desc, because we can only describe whether a regex pattern matches or not. Can we describe how many times it matches?

> Why is this change being made at all?

Inclusion checks are sloooow